### PR TITLE
Use gerrit.w.o instead of github for wikibase.

### PR DESCRIPTION
**Pull request by JanZerebecki**
The mirror on github is lagging behind.

### CODE REVIEW

**Comment by JanZerebecki** [comment]
However this might surface the problem worked around in #17 again.

File: composer.json [hunk 1]
```diff
@@ -10,6 +10,18 @@
         {
             "type": "vcs",
             "url": "https://github.com/wmde/WikimediaBadges.git"
+        },
+        {
+            "type": "package",
+            "package": {
```
Comment (JanZerebecki) [inline composer.json]: This does not pull in recursive dependencies, because this should be the full composer.json for one version. Use type:vcs instead.

**Comment by tobijat** [comment]
Made a new build today using this change and it worked. It's slow and timed out several times (most likely due to cloning Wikibase from Gerrit (twice?)) but finally it went through. Also doing "export COMPOSER_PROCESS_TIMEOUT=600" would prevent timeouts.
Btw. the GitHub mirror issue got fixed yesterday, that brings us to the question, do we want to use the mirror or clone from gerrit?

**Comment by tobijat** [comment]
@filbertkm @mariushoch opinions?

**Comment by mariushoch** [comment]
As long as we wont see these github replication disruptions more often I tend towards using github as it's generally faster and probably more stable. This is the first time we had such issues AFAIK.

Also I'm not sure how this is going to affect what I did in #17 .

**Comment by mariushoch** [comment]
Breaks updating the build for me, will undo for 1.25wmf16. Consider this a -2.

```  [RuntimeException]                                                                                                               
  Failed to execute git clone --mirror 'https://git.wikimedia.org/git/mediawiki/extensions/Wikibase.git' '/home/hoch_m/.composer/  
  cache/vcs/https---git.wikimedia.org-git-mediawiki-extensions-Wikibase.git/'                                                      
  Cloning into bare repository '/home/hoch_m/.composer/cache/vcs/https---git.wikimedia.org-git-mediawiki-extensions-Wikibase.git'  
  ...                                                                                                                              
  error: RPC failed; result=22, HTTP code = 411                                                                                    
  fatal: The remote end hung up unexpectedly ```

**Comment by JanZerebecki** [comment]
That is a transient failure of the connection. But how did it get from gerrit.w.o to git.w.o ?

**Comment by mariushoch** [comment]
Maybe gerrit redirects those URLs to git.w.o? Anyway I think just using gerrit is fine and probably even more stable than gerrit/ gitblit.

**Comment by filbertkm** [comment]
if something on packagist or github is not working when i am making a build (or i don't want to wait for packagist), I will add the vcs directly in the build.

since we have so many dependencies on github (e.g. data values), we still depend on it at the moment.

**Comment by filbertkm** [comment]
also, what's the reason for "type": "git" instead of "type": "vcs"?

**Comment by JanZerebecki** [comment]
There is no .git at the end of the URL so the auto detection from vcs does not work.

**Comment by JanZerebecki** [comment]
Anyway someone can revive this once it becomes more important or more parts of moving away from github are solved.

**Pull request closed without merging**